databricks/koalas/tests/test_dataframe.py

```diff
@@ -2396,6 +2396,15 @@ def test_reindex(self):
             kdf.reindex(columns=["numbers", "2", "3"], fill_value=0.0).sort_index(),
         )
 
+        index2 = ["A", "C", "D", "E", "0"]
+        pdf2 = pd.DataFrame({"a": [1.0, 2.0, 3.0, 4.0, 5.0]}, index=index2)
+        kdf2 = ks.from_pandas(pdf2)
+
+        self.assert_eq(
+            pdf.reindex(index=pdf2.index, fill_value=0.0).sort_index(),
+            kdf.reindex(index=kdf2.index, fill_value=0.0).sort_index(),
+        )
```

Comment:
Also what if the given index is `MultiIndex`?

Reply:
Thanks for noticing, I updated PR so it has the same behavior of pandas, considering we currently can only reindex single level index.